### PR TITLE
Fix issue where $refs were not resolved within lists

### LINKED ISSUE #6: References not resolved within lists
References nested with in a list will not be resolved. For example, the following YAML code:
```
paths:
  /batches:
    post:
      summary: Sends a BatchList to the validator
      consumes:
        - application/octet-stream
      parameters:
        -
          name: BatchList
          in: body
          description: A binary encoded protobuf BatchList
          schema:
            $ref: "#/definitions/BatchList"
          required: true
        -
          $ref: "#/parameters/callback"
```

Neither reference will be caught by `_resolve_refs`.

### PR DESCRIPTION
**Pull request by delventhalz**
Fixes #6

Signed-off-by: Zac Delventhal <zac@bitwise.io>

### CODE REVIEW

File: sphinxcontrib/openapi.py [hunk 1]
```diff
@@ -57,15 +57,17 @@ def _resolve_refs(uri, spec):
...
-            elif isinstance(v, collections.Mapping):
+        if isinstance(node, collections.Mapping) and '$ref' in node:
+            with resolver.resolving(node['$ref']) as resolved:
+                return resolved
+        elif isinstance(node, collections.Mapping):
+            for k, v in node.items():
                 node[k] = _do_resolve(v)
-            else:
-                node[k] = v
+        elif isinstance(node, list):
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi.py]: Could you please use `collections.Sequence` instead of `list`?

**Comment by delventhalz** [comment]
I have not tested the fix thoroughly, but it works with the Open API spec I'm currently compiling. I have attached it.

[openapi.yml.zip](https://github.com/ikalnytskyi/sphinxcontrib-openapi/files/720646/openapi.yml.zip)

**Comment by delventhalz** [comment]
Do not merge yet. Switching `list` to `collections.Sequence` causes an infinite loop. I will look into it.

**Comment by ikalnytskyi** [comment]
> Do not merge yet. Switching list to collections.Sequence causes an infinite loop. I will look into it.

Ah, I believe the reason is that `str` could be considered as `collections.Sequence`. Maybe then it's really better to use `list` or even `(list, tuple)` instead.

**Comment by delventhalz** [comment]
> Ah, I believe the reason is that str could be considered as collections.Sequence. Maybe then it's really better to use list or even (list, tuple) instead.

Yeah, strings are the problem. `collections.MutableSequence` would skip the strings but it would also skip tuples. I've updated it to `isinstance(x, list, tuple)`. That seems to make the most sense.

**Comment by delventhalz** [comment]
Hey ikalnytskyi! I'm wondering what the likelihood is of my two PRs getting merged and published to pip, and if so, what would the ETA of that be? Your tool is great, exactly what I need for a project at work, but we need these two changes incorporated before we can use it.

**Comment by ikalnytskyi** [comment]
@delventhalz Sorry for the delay. Thanks you for working on this, I really eager to merge these PRs, but I don't merge anything without tests (if you take a look onto one of my reviews, I asked you add one).

I thought you're going to update it, but if you don't like, would you mind if I pickup your work and push a test for you?

**Comment by delventhalz** [comment]
Ah, I see. I didn't understand you were asking me to write tests. I will add those to the PR today or tomorrow.

**Comment by delventhalz** [comment]
Okay, tests have been added/updated for both updates.

**Comment by ikalnytskyi** [comment]
@delventhalz thank you ver much!

**Comment by ikalnytskyi** [comment]
@delventhalz I've pushed version 0.3.0 :)

https://pypi.org/project/sphinxcontrib-openapi/0.3.0/

**Comment by delventhalz** [comment]
Thank you very much!

**Pull request merged**